Docs: Edit/move "Installation" → "Getting Started"

medubelko opened the pull request:
First part of migrating introductory material under new "Getting Started" section. Introductory tutorials will be housed here in later PRs.

#### Changes
- Create "Getting Started" section.
- Rewrite and expand "Installation" into separate installation and configuration articles:
    - "Installing Gaffer"
    - "Launching Gaffer for the First Time"
    - "Setting Up the Gaffer Command"
    - "Configuring Gaffer for Third-Party Tools"
- Remove "Installation".

medubelko commented on doc/source/GettingStarted/ConfiguringGafferForThirdPartyTools/index.md:
```diff
@@ -0,0 +1,196 @@
+    <!-- TODO: ![Delight node menu](images/delightNodes.png) -->
+
+
+## Configuring Gaffer for Tractor ##
```
Andrew mentioned that only users at studios would need this information. Would it be prudent to move this into the "Configuring Gaffer at Your Studio" section?

johnhaddon replied:
I would leave it in the third party tools section, because Tractor is a third party tool, and it's not necessarily limited to studios either (I presume that free RenderMan comes with it).

medubelko commented on doc/source/conf.py:
```diff
@@ -310,6 +310,20 @@
 
+# Variables for string replacement functions
+
+arnold_version = '2018'
```
Standalone isn't even offered on their trial page... I was taking an approach for individuals (as you mentioned, TDs/studios would likely be able to fill in the blanks). I figured the most common use-case would be artists using the Maya version.

andrewkaufman replied:
Its on this page under the "Arnold SDK" section:
https://www.solidangle.com/arnold/download/

That's buried way down that page, but I don't see why we'd be suggesting people download the Maya plugin, especially as we don't really want them loading the maya-specific bits of it into Gaffer.

Also, its really confusing to be thinking "I want Arnold 5.1" and seeing `arnold_version = '2018'`

medubelko commented on doc/source/conf.py:
```diff
@@ -310,6 +310,20 @@
+arnold_path_linux = '/opt/solidangle/mtoa/{0}'.format( arnold_version )
+arnold_path_osx = '/opt/solidangle/mtoa/{0}'.format( arnold_version )
+
+delight_version = '8.5.1'
```
Hmm, another trial inconsistency. Latest free public standalone is 12.5.9, latest free Maya version is 8.5.1. Again, it depends on the most likely use-case.

medubelko commented on doc/source/GettingStarted/SettingUpGafferCommand/index.md:
```diff
@@ -0,0 +1,142 @@
+# Setting Up the "gaffer" Command #
```
It's a compromise: ideally, I'd use the inline code tag, but that messes with the titles in the TOC, hence the quotation marks. 

I want to reduce confusion between Gaffer the app, API and node classes that start with the word Gaffer, and commands. It helps with skimming. Since the command is lowercase (as an env var it could be anything, but the script itself is `bin/gaffer`, so I'm sticking to the most 1:1 approach), and case matters in Linux, it's more accurate this way.

andrewkaufman replied:
Oh, sorry, I didn't mean capitalization of "gaffer", I meant of Up and Command

medubelko replied:
I'm using title case, mostly out of habit. You'll see a lot of web content use sentence case, but title case is still fairly popular in more formal writing. I'm not married to either form, and if you prefer I could make a PR undoing all of them (after we merge my PRs – all my H1s and H2s are like this, and rebasing everything at this stage will be a pain). Of possible note: I've been using it everywhere so far, so I wonder if you noticed it this time because it was juxtaposed with "gaffer"?

medubelko commented on doc/source/GettingStarted/SettingUpGafferCommand/index.md:
````diff
@@ -0,0 +1,142 @@
+MacBook:~ user$ gaffer
+```
+
+## Opening Gaffer Scripts in the Terminal ##
````
Yeah, we should link to the startup parameters.

medubelko commented on doc/source/GettingStarted/SettingUpGafferCommand/index.md:
```diff
@@ -0,0 +1,142 @@
+
+1. Open a terminal.
+
+2. Open your terminal's user configuration file in a text editor such as _vim_, _nano_, or _gedit_.
```
Heh, I thought I'd throw a bone to those who insist on using it, but I can take it out.

medubelko commented on doc/source/GettingStarted/SettingUpGafferCommand/index.md:
```diff
@@ -0,0 +1,142 @@
+> For these instructions, we will assume you have Gaffer installed to the `/opt/` directory. If you have installed it elsewhere, replace `/opt/` with the directory you installed it to.
+
+
+### Environment Variables ###
```
Here's what I said to John about this:

> In my experience (interacting with users at my old jobs, as well as talking to artists here) there are very, very few day-to-day (a.k.a. "rote") users who would immediately react to the words "environment variable" by thinking "I know what those are and what is involved in setting them up." I asked IE's LookDev lead what he thought of configuring environment variables, and he admitted it was foreign territory for him. That, combined with finding a pretty abysmal lack of coverage by the Linux community on how to _permanently_ set up an env var that didn't devolve into navel-gazing debates on which shell configuration file to use, motivated my fleshing out that article.

So, yeah, not super revelatory to other studio techs, but probably helpful to people trying it out on their home machine. This will become a moot point once we have an interactive installer, but I don't think it's actively harmful to spent two paragraphs establishing what we're talking about before prescribing to the public to mess with their env vars.

medubelko commented on doc/source/GettingStarted/ConfiguringGafferForThirdPartyTools/index.md:
```diff
@@ -0,0 +1,194 @@
+    <!-- TODO: ![Arnold node menu](images/arnoldNodes.png) -->
+
+
+### Arnold in OSX ###
```
The section is indeed the same. This is part of the awkwardness of splitting style and content. Once the tabs are in, it will make more sense. I could comment it out for now, I suppose.

The Finder part is duplicated because documentation is never read serially. We can't assume whoever lands here will have read the articles leading up to it. There will be plenty more redundancy as this progresses...

medubelko commented on doc/source/GettingStarted/ConfiguringGafferForThirdPartyTools/index.md:
```diff
@@ -0,0 +1,194 @@
+## See Also ##
+
+- [Setting Up the "gaffer" Command](../SettingUpGafferCommand/index.md)
+- [Installing Gaffer](../InstallingGaffer/index.md)
```
There aren't any other third-party related articles yet. Would going back to the starter tutorial be sufficient?

andrewkaufman replied:
How about a suggesting they go back to the starter tutorial and replace the Appleseed nodes with Arnold/3delight nodes?

andrewkaufman commented on doc/source/GettingStarted/SettingUpGafferCommand/index.md:
```diff
@@ -0,0 +1,142 @@
+3. At the end of the file, append the location of Gaffer's binary directory to the `PATH` variable:
+    
+    - _bash_: `export PATH=$PATH\:/opt/gaffer-!GAFFER_VERSION!-linux/bin`
+    - _tcsh_: `setenv PATH=$PATH\:/opt/gaffer-!GAFFER_VERSION!-linux/bin`
```
that's not the error. tcsh doesn't use `=` in `setenv`

andrewkaufman commented on doc/source/GettingStarted/SettingUpGafferCommand/index.md:
```diff
@@ -0,0 +1,142 @@
+
+4. Save the file and close the text editor.
+
+5. In the terminal, reload the user configuration file and test that the `PATH` variable has been updated:
```
Then I'd lean towards spawning a child terminal by running `tcsh` or `bash` from the same terminal I guess. But we can leave it as sourceing...

andrewkaufman commented on doc/source/GettingStarted/ConfiguringGafferForThirdPartyTools/index.md:
````diff
@@ -0,0 +1,194 @@
+    !ARNOLD_PATH_LINUX!
+    ```
+
+4. The next time you start Gaffer, the Arnold nodes will be available from the node creation menu.
````
K, cool. FYI I think it'd be simple to add a Title to the menu in the interface if that'd help. Not sure how people will react to the aesthetics of that though...

medubelko commented:
Implemented suggested changes and refined some of the duplicated language.

Updated the local preview: `/home/michaeldu/dev/gafferDocsBuilds/docsInstallationEdit`

medubelko commented on doc/source/GettingStarted/ConfiguringGafferForThirdPartyTools/index.md:
```diff
@@ -55,17 +50,21 @@ To create the `ARNOLD_ROOT` environment variable in OSX:
+
+The next time you start Gaffer, the Arnold nodes will be available from the node creation menu (right-click inside the _Graph Editor_).
+
+<!-- TODO: ![Arnold node menu](images/arnoldNodes.png) -->
```
@andrewkaufman How should we handle the 3rd party screenshots? Add a screengrab.py to gafferResources, then have them copied over?

medubelko commented:
Rebased.

johnhaddon commented:
@andrewkaufman, you happy to merge this?
